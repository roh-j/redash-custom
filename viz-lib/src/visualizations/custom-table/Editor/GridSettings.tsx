import React from "react";
import { EditorPropTypes } from "@/visualizations/prop-types";
import { map } from "lodash";
import { Section, Select } from "@/components/visualizations/editor";

const ALLOWED_ITEM_PER_PAGE = [5, 10, 15, 20, 25, 50, 100, 150, 200, 250, 500];

export default function GridSettings({ options, onOptionsChange }: any) {
  return (
    // @ts-expect-error ts-migrate(2745) FIXME: This JSX tag's 'children' prop expects type 'never... Remove this comment to see the full error message
    <Section>
      <Select
        label="Items per page"
        defaultValue={options.itemsPerPage}
        onChange={(itemsPerPage: any) => onOptionsChange({ itemsPerPage })}>
        {map(ALLOWED_ITEM_PER_PAGE, value => (
          // @ts-expect-error ts-migrate(2339) FIXME: Property 'Option' does not exist on type '({ class... Remove this comment to see the full error message
          <Select.Option key={`ipp${value}`} value={value}>
            {value}
            {/* @ts-expect-error ts-migrate(2339) FIXME: Property 'Option' does not exist on type '({ class... Remove this comment to see the full error message */}
          </Select.Option>
        ))}
      </Select>
    </Section>
  );
}

GridSettings.propTypes = EditorPropTypes;
