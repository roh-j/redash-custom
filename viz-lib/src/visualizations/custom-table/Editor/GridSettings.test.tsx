import React from "react";
import enzyme from "enzyme";

import getOptions from "../getOptions";
import GridSettings from "./GridSettings";

function findByTestID(wrapper: any, testId: any) {
  return wrapper.find(`[data-test="${testId}"]`);
}

function mount(options: any, done: any) {
  const data = { columns: [], rows: [] };
  options = getOptions(options, data);
  return enzyme.mount(
    <GridSettings
      visualizationName="Test"
      data={data}
      options={options}
      onOptionsChange={changedOptions => {
        expect(changedOptions).toMatchSnapshot();
        done();
      }}
    />
  );
}

describe("Visualizations -> Custom Table -> Editor -> Grid Settings", () => {
  test("Changes items per page", done => {
    const el = mount(
      {
        itemsPerPage: 25,
      },
      done
    );

    findByTestID(el, "CustomTable.ItemsPerPage")
      .last()
      .simulate("mouseDown");
    findByTestID(el, "CustomTable.ItemsPerPage.100")
      .last()
      .simulate("click");
  });
});
