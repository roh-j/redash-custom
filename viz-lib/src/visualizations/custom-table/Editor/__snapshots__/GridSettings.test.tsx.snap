// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualizations -> Custom Table -> Editor -> Grid Settings Changes items per page 1`] = `
Object {
  "itemsPerPage": 100,
}
`;
